chronos/asyncmacro2.nim

```diff
@@ -48,21 +27,33 @@ proc completeWithNode(fut, baseType, node: NimNode): NimNode {.compileTime.} =
           ident "is", nnkTypeOfExpr.newTree(node), ident "void"),
         newStmtList(
           node,
-          completeWithResult(fut, baseType)
         )
       ),
       nnkElseExpr.newTree(
-        newCall(ident "complete", fut, node)
+        newAssignment(ident "result", node)
```

Comment:
this should directly use `internalValue` rather than going via the template (more templates = more problems)

Reply:
and processBody..

Reply:
This proc is only used with `procBodyBlck` which is already went through `processBody`

Reply:
right

Reply:
> more templates = more problems

That's unfortunately true, since using internalValue instead of the template fixed some compilation issue in nimbus
Looks like something that might bite us later